Add Nuget Support

LukeWinikates opened the pull request:
This PR adds basic support for the Nuget package manager as described in #155 . Dependency group support is in. License discovery will wait until later. For `LicenseFinder` will discover all the package dependencies for each assembly, and users will use `license_finder licenses add` to set the licenses manually.

mainej commented:
Looks like a good start.  Thanks for starting this.  Check out [how the rebar package manager was added](https://github.com/pivotal/LicenseFinder/compare/v2.0.0...v2.0.1).  If possible, please add feature specs and travis config, and update the CHANGELOG, README and Rakefile.  It would be nice if you mention the limitations around setting licenses in the README, at least until you implement the find-license-by-url idea.

LukeWinikates commented:
Thanks @mainej -- I've added the feature spec and updated documentation. Because there's no dependency on a command line tool, I don't think there's anything to add to the Rakefile or special travis config.

mainej commented:
:+1: Thanks!
